src/components/magnet/magnet-canvas.tsx

```diff
@@ -96,16 +96,29 @@ export class MagnetCanvas extends BaseComponent<IProps, IState> {
 
   public render() {
     const {simulation} = this.stores;
+
     const primaryMag = simulation.getMagnetAtIndex(0);
     const primaryMagType: MagnetType | undefined = primaryMag ? primaryMag.type : undefined;
-    const flip1 = primaryMag && primaryMag.barPolarity === "S-N" ? true : false;
+    const primaryFlip = primaryMag && primaryMagType === "bar" ? primaryMag.flipped : false;
+    const primaryImage = primaryMag ? primaryMag.magnetImage : "";
+    const primaryImageOffset: number = primaryMag ? primaryMag.polarityCurrentImageOffset : 0;
+    const primaryLeftPoleImage: string = primaryMag ? primaryMag.leftPoleImage : "";
+    const primaryRightPoleImage: string = primaryMag ? primaryMag.rightPoleImage : "";
+    const primaryCurrentArrowImage: string = primaryMag ? primaryMag.currentArrowImage : "";
+    const primaryVoltageImage: string = primaryMag ? primaryMag.voltageImage : "";
+    const primaryVoltageFlip: boolean = primaryMag ? primaryMag.voltageFlip : false;
+
     const secondaryMag = simulation.getMagnetAtIndex(1);
     const secondaryMagType: MagnetType | undefined = secondaryMag !== null ? secondaryMag.type : undefined;
-    const flip2 = secondaryMag && secondaryMag.barPolarity === "S-N" ? true : false;
-    const primaryImage = primaryMag ? primaryMag.magnetImage : "";
+    const secondaryFlip = secondaryMag && secondaryMagType === "bar" ? secondaryMag.flipped : false;
     const secondaryImage = secondaryMag ? secondaryMag.magnetImage : "";
-    const primaryLength = primaryMag ? primaryMag.magnetLength : 0;
-    const secondaryLength = secondaryMag ? secondaryMag.magnetLength : 0;
+    const secondaryImageOffset: number = secondaryMag ? secondaryMag.polarityCurrentImageOffset : 0;
+    const secondaryLeftPoleImage: string = secondaryMag ? secondaryMag.leftPoleImage : "";
+    const secondaryRightPoleImage: string = secondaryMag ? secondaryMag.rightPoleImage : "";
+    const secondaryCurrentArrowImage: string = secondaryMag ? secondaryMag.currentArrowImage : "";
+    const secondaryVoltageImage: string = secondaryMag ? secondaryMag.voltageImage : "";
+    const secondaryVoltageFlip: boolean = secondaryMag ? secondaryMag.voltageFlip : false;
```

Comment:
This should all get cleaned up a bunch when we can just pass the magnet model down to the magnet component.

Reply:
Agreed!